sumpy/e2e.py

```diff
@@ -332,7 +332,12 @@ def get_translation_loopy_insns(self, result_dtype):
                 [sym.Symbol(f"data{i}")
             for i in range(m2l_translation_classes_dependent_ndata)]
 
-        ncoeff_src = len(self.src_expansion)
+        m2l_translation = self.tgt_expansion.m2l_translation
+        if m2l_translation.use_preprocessing:
+            ncoeff_src = m2l_translation.preprocess_multipole_nexprs(
+                self.tgt_expansion, self.src_expansion)
+        else:
+            ncoeff_src = len(self.src_expansion)
```

Comment:
Maybe `ncoeff_src` should be renamed to clarify that we're talking about "padded" coeffiicents. (Though I'm not sure "padded" is a good adjective.)